br/pkg/lightning/backend/local/engine.go

```diff
@@ -235,6 +235,18 @@ func (e *Engine) unlock() {
 	e.mutex.Unlock()
 }
 
+func (e *Engine) TotalMemorySize() int64 {
+	var memSize int64 = 0
+	e.localWriters.Range(func(k, v interface{}) bool {
+		w := k.(*Writer)
+		if w.kvBuffer != nil {
+			memSize += w.kvBuffer.TotalSize()
```

Comment:
`kvBuffer` may not be a thread-safe object.

Reply:
Added aquire/release writer lock to make it thread safe.

Reply:
Added writer lock/unlock logic.